packages/manager/apps/zimbra/src/pages/dashboard/EmailAccounts/ModalAddAlias.tsx

```diff
@@ -0,0 +1,273 @@
+  const [searchParams] = useSearchParams();
+  const editEmailAccountId = searchParams.get('editEmailAccountId');
+  const [isLoading, setIsLoading] = useState(true);
+  const { addError, addSuccess } = useNotifications();
+  const navigate = useNavigate();
+  const goBackUrl = useGenerateUrl('..', 'path', { editEmailAccountId });
+  const goBack = () => navigate(goBackUrl);
+
+  type FieldType = {
+    value: string;
+    hasError?: boolean;
+    required?: boolean;
+  };
+
+  interface FormTypeInterface {
+    [key: string]: FieldType;
+  }
+
+  interface FormInputRegexInterface {
+    [key: string]: RegExp;
+  }
+
+  const formInputRegex: FormInputRegexInterface = {
+    alias: /^(?:[A-Za-z0-9]+(?:[-_][A-Za-z0-9]+)*)(?:(?:[.|+])(?:[A-Za-z0-9]+(?:[-_][A-Za-z0-9]+)*))*$/,
+  };
+
+  const [form, setForm] = useState<FormTypeInterface>({
+    alias: {
+      value: '',
+      hasError: false,
+      required: true,
+    },
+    domain: {
+      value: '',
+      required: true,
+    },
+  });
+
+  const [isFormValid, setIsFormValid] = useState(false);
+
+  const { data: domainList, isLoading: isLoadingDomain } = useDomains();
+
+  const { data: editAccountDetail, isLoading: isLoadingEmailDetail } = useQuery(
+    {
+      queryKey: getZimbraPlatformAccountDetailQueryKey(
+        platformId,
+        editEmailAccountId,
+      ),
+      queryFn: () =>
+        getZimbraPlatformAccountDetail(platformId, editEmailAccountId),
+      enabled: !!platformId && !!editEmailAccountId,
+    },
+  );
+
+  useEffect(() => {
+    if (!isLoadingDomain && !isLoadingEmailDetail && platformId) {
+      setIsLoading(false);
+    }
+  }, [isLoadingDomain, isLoadingEmailDetail]);
+
+  const handleNewAliasClick = () => {
+    const {
+      alias: { value: alias },
+      domain: { value: domain },
+    } = form;
+
+    const dataBody = {
+      alias: `${alias}@${domain}`,
+      aliasTarget: editAccountDetail?.currentState.email,
+      organizationId: editAccountDetail?.currentState.organizationId,
+    };
+
+    postZimbraPlatformAlias(platformId, dataBody)
+      .then(() => {
+        goBack();
+        addSuccess(
+          <OsdsText
+            color={ODS_THEME_COLOR_INTENT.text}
+            size={ODS_THEME_TYPOGRAPHY_SIZE._100}
+            level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
+            hue={ODS_THEME_COLOR_HUE._500}
+          >
+            {t('zimbra_account_alias_add_success_message')}
+          </OsdsText>,
+          true,
+        );
+      })
+      .catch(({ response }) => {
+        goBack();
+        addError(
+          <OsdsText
+            color={ODS_THEME_COLOR_INTENT.text}
+            size={ODS_THEME_TYPOGRAPHY_SIZE._100}
+            level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
+            hue={ODS_THEME_COLOR_HUE._500}
+          >
+            {t('zimbra_account_alias_add_error_message', {
+              error: response.data.message,
+            })}
+          </OsdsText>,
+          true,
+        );
+      });
```

Comment:
**nitpick:** Handle this tanstack query mutation.

Reply:
It's applied